accepted/2021/experimental-features/experimental-features.md

```diff
@@ -0,0 +1,380 @@
+
+### Enabling a cross-cutting experimental feature
+
+Ainsley is working for the insurance company Fabrikam. As part of their work,
+they are trying to figure out how they can integrate machine learning with
+ML.NET more tightly with their services, but they struggle with data ingestion
+because a lot of the heavy computations are happening inside of an old C++ base.
+When that code was written, .NET wasn't fast enough to do heavy numerical
+computations.
+
+As part of their research into this problem space, they notice a recent blog
+post about how the upcoming generic numeric support and hardware intrinsics can
+help to speed things up considerably. They decide to give it a spin, so they
+copy & paste some sample code from the blog post.
+
+The code immediately produces an error message:
+
+> error: The type 'IArithmetic\<T>' is part of the experimental feature
+> 'StaticInterfaceMembers' which must be turned on explicitly.
+
+After taking a quick look at the blog post, Ainsley realizes that they only need
+to set the `ExperimentalFeatures` property in the project files.
+
+### Consuming a library that uses an experimental feature
+
+Sahibi works with Ainsley and tries to consume their library that exposes some
+of the core algorithms so she can integrate those with ML.NET. After adding the
+reference to Ainsley's library `Fabrikam.Algorithms.Core` she instantiates the
+engine type which produces the following error:
+
+> The library 'Fabrikam.Algorithms.Core' requires the experimental feature
+> 'StaticInterfaceMembers' which must be turned on explicitly.
+
+Sahibi IMs Ainsley to ask what this is about where she learns that this feature
+isn't a supported part of .NET, therefore she should make sure to not integrate
+the library into the production instance of Fabrikam.
+
+## Requirements
+
+### Goals
+
+* Customers can discover experimental features
+* Customers are made aware if a particular feature is in an experimental state
+  so that they always know which part of the product is fully supported and
+  which parts aren't.
+* Customer have an easy way to audit if their applications or libraries are
+  using experimental features.
+* Customers can't accidentally depend on experimental features, either directly
+  or indirectly through some library.
+* We can deliver a reasonable experience even when an experimental feature can't
+  be supported in all the workloads, for example, by failing with a sensible
+  error message.
+* We have a unified experience for turning on experimental features, no matter
+  whether they are in a runtime, a compiler, a library, or span multiple parts
+  of the system.
+* We don't have to do unnatural acts to add an experimental feature, by, for
+  example, having to build a plug-in based system that allows swapping in and
+  out files.
+
+### Non-Goals
+
+* Allowing Microsoft to turn features on or off remotely, e.g. A/B testing
+* Designing a mechanism for shipping experimental features as separate packages
```

Comment:
It would be helpful to understand how this works with https://github.com/dotnet/runtimelab.